src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Mapping/FacetMappingProfile.cs

```diff
@@ -13,41 +13,45 @@ public FacetMappingProfile()
                 {
                     "attr" => new TermFacetResult
                     {
-                        Terms = request.Items.Select(x => new FacetTerm
-                            {
-                                Count = x.Count,
-                                IsSelected = x.IsApplied,
-                                Term = x.Value.ToString(),
-                                Label = x.Labels != null ? string.Join(' ', x.Labels.Select(l => l.Label)) : x.Value.ToString(),
-                            })
+                        Terms = request.Items.Select(x => new FacetTerm
+                        {
+                            Count = x.Count,
+                            IsSelected = x.IsApplied,
+                            Term = x.Value.ToString(),
+
+                            Label = x.Labels?.FirstOrDefault(x => x.Language == cultureName)?.Label ?? x.Value.ToString(),
+                        })
                             .ToArray(),
                         Name = request.Field
                     },
                     "pricerange" => new RangeFacetResult
                     {
-                        Ranges = request.Items.Select(x => new FacetRange
-                            {
-                                Count = x.Count,
-                                From = Convert.ToInt64(x.RequestedLowerBound),
-                                IncludeFrom = !string.IsNullOrEmpty(x.RequestedLowerBound),
-                                FromStr = x.RequestedLowerBound,
-                                To = Convert.ToInt64(x.RequestedUpperBound),
-                                IncludeTo = !string.IsNullOrEmpty(x.RequestedUpperBound),
-                                ToStr = x.RequestedUpperBound,
-                                IsSelected = x.IsApplied,
-                                Label = x.Value.ToString(),
-                            })
+                        Ranges = request.Items.Select(x => new FacetRange
+                        {
+                            Count = x.Count,
+                            From = Convert.ToInt64(x.RequestedLowerBound),
+                            IncludeFrom = !string.IsNullOrEmpty(x.RequestedLowerBound),
+                            FromStr = x.RequestedLowerBound,
+                            To = Convert.ToInt64(x.RequestedUpperBound),
+                            IncludeTo = !string.IsNullOrEmpty(x.RequestedUpperBound),
+                            ToStr = x.RequestedUpperBound,
+                            IsSelected = x.IsApplied,
+                            Label = x.Value.ToString(),
+                        })
                             .ToArray(),
                         Name = request.Field,
                     },
                     _ => null
                 };
-
+                if (result != null)
+                    result.Label = request.Labels?.FirstOrDefault(x => x.Language == cultureName)?.Label ?? result.Name;

```

Comment:
```suggestion

                if (result != null)
                    result.Label = request.Labels?.FirstOrDefault(x => x.Language == cultureName)?.Label ?? result.Name;
                    
```

Reply:
All logic for selecting a localized string for a given culture  must be reused from a single place 

https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/VirtoCommerce.ExperienceApiModule.Core/Extensions/IHasLanguageExtensions.cs

You have to extend this extension method to be able to work from object types that not implement `IHasLanguage` interface.
```C#
 public static IHasLanguage FirstBestMatchForLanguage(this IEnumerable hasLanguages, Func<object, string> langSelector, string language)
```